client/asset/dcr/dcr.go

```diff
@@ -2137,6 +2137,16 @@ func (dcr *ExchangeWallet) lockedAtoms() (uint64, error) {
 	return sum, nil
 }
 
+// getTxOut attempts to find the specified txout from the regular tree and if
+// not found in the regular tree, checks the stake tree.
+func (dcr *ExchangeWallet) getTxOut(txHash *chainhash.Hash, index uint32, mempool bool) (*chainjson.GetTxOutResult, error) {
+	txout, err := dcr.node.GetTxOut(dcr.ctx, txHash, index, wire.TxTreeRegular, mempool) // check regular tree first
+	if err == nil && txout == nil {
+		txout, err = dcr.node.GetTxOut(dcr.ctx, txHash, index, wire.TxTreeStake, mempool) // check stake tree
+	}
```

Comment:
I remember the conversation about the (mostly theoretical) potential for there to exist 2 txouts with same hash and vout but different trees. Doesn't that make this code potentially problematic, as the first regular tree check may return a result but not what we really wanted?

Might it be better to modify the coin id encoding for dcr to allow specifying the tree rather than trying to figure it out for ourselves?

Reply:
That's conceivable for 0.3/1.0 after we've had asset versioning in place for a while, but it's a major challenge.
My takeaway from previous discussions is that there is no risk with just checking both trees:

https://matrix.to/#/!SFRQQFIHUUNXARfvew:decred.org/$ZK-TYu3aSpJmtd3VOEFhCMDExIBo6hciHOw7oDoGGWk?via=decred.org&via=matrix.org&via=planetdecred.org

> All of that said, as I mentioned above, in practice, it's virtually impossible to have a collision.
> 
> So, from a purely technical sense, a collisions would be an issue.  However, your takeaway is correct.
>
> This is why I wanted rstaudt2 to ask though, because it sounds like the dex protocol doesn't have the information, so either gettxout does the check in both places (as it does now) or the caller does.

...

> I should note that in the (probably never going to happen in practice) aforementioned case where it ends up referencing a collision in the wrong tree, because dcrd internally handles it properly, trying to spend it would end up with a rejected transaction because it's part of the signature hash.
> So, I don't believe there is any real danger here outside of perhaps not handling that case properly by trying the other tree.
>
> Also, you can't include a regular tx in the stake tree, nor vice versa, so to intentionally create a collision, you would have to break the hash functions preimage resistance, and if that happens, there are much bigger problems